### PR TITLE
added functionality to generate article url which fixes #464

### LINKED ISSUE #464: article url generation
I think urls of articles should be generated automatically or by clicking a button specifically for that like checking the availability of a url, to prevent manually adding it since some articles could have  lengthy urls

### PR DESCRIPTION
**Pull request by JahlomP**

### CODE REVIEW

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html [hunk 1]
```diff
@@ -35,6 +36,21 @@
 			});
 		};
 
+		$scope.generateUrl = function() {
+			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
```
Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: This can be it's own function that returns the generated URL, so we don't need to have the regex defined twice.

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html [hunk 2]
```diff
@@ -35,6 +36,21 @@
...
+			if($scope.article.url){
+				$scope.urlGenerated = true;
+			} else {
+				$scope.urlGenerated = false;
+			}
+		};
+
+		$scope.$watch('article.url', function() {
```
Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: ```$scope.$watch('article.url', function(newVal, oldVal)...```

Then you can check if the newVal is different than the oldVal before doing a string compare on the generated URL.

File: plugins/pencilblue/templates/admin/content/articles/article_form.html [hunk 3]
```diff
@@ -7,12 +7,26 @@
 		<fieldset>
 			<div class="tab-content">
 				<div class="tab-pane active" id="content">
+					<div class="form-group" ng-class="{'has-error': !isFieldValid(articleForm.headline)}">
```
Comment (blakecallens) [inline plugins/pencilblue/templates/admin/content/articles/article_form.html]: Why did you move this above the URL?

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html [hunk 4]
```diff
@@ -35,6 +36,21 @@
 			});
 		};
 
+		$scope.generateUrl = function() {
+			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
+			if($scope.article.url){
+				$scope.urlGenerated = true;
+			} else {
```
Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: We'll never have this hit, if the headline is valid. Also, I would like to make sure the URL is unique/valid. You can make a call to the check function at this point to make sure of that.

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html (hunk 1 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: Will work on that soon

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html (hunk 2 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: Will work on this soon

File: plugins/pencilblue/templates/admin/content/articles/article_form.html (hunk 3 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/admin/content/articles/article_form.html]: Since url will be generated from the headline, it is only natural to first define the headline then use it in url generation.

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html (hunk 4 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: Will do the url check but what do you mean by "We'll never have this hit, if the headline is valid."?

File: plugins/pencilblue/templates/admin/content/articles/article_form.html (hunk 3 above)
Comment (blakecallens) [inline plugins/pencilblue/templates/admin/content/articles/article_form.html]: I'm fine with that, but it'll need to change on the page form too, for consistency.

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html (hunk 4 above)
Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: The generate button is only active if there's a valid headline, which means that a url will always be generated.

File: plugins/pencilblue/templates/admin/content/articles/article_form.html (hunk 3 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/admin/content/articles/article_form.html]: K

File: plugins/pencilblue/templates/angular/admin/content/articles/article_form.html (hunk 4 above)
Comment (JahlomP) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: So should I use the length option?

Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: ```if($scope.article.url && $scope.article.url.length)```

protects against both null and empty string

Comment (JahlomP) [inline plugins/pencilblue/templates/angular/admin/content/articles/article_form.html]: OK. Thanks.

On Tue, Nov 10, 2015 at 7:01 PM Blake Callens <notifications@github.com>
wrote:

> In
> plugins/pencilblue/templates/angular/admin/content/articles/article_form.html
> <https://github.com/pencilblue/pencilblue/pull/810#discussion_r44448770>:
>
> > @@ -35,6 +36,21 @@
> >  			});
> >  		};
> >
> > +		$scope.generateUrl = function() {
> > +			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
> > +			if($scope.article.url){
> > +				$scope.urlGenerated = true;
> > +			} else {
>
> if($scope.article.url && $scope.article.url.length)
>
> protects against both null and empty string
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pencilblue/pencilblue/pull/810/files#r44448770>.
>
-- 
Best Rregards,
Precious Jahlom Agboado,
www.twitter.com/jahlomp
+233 54 0915 108.

**Comment by JahlomP** [comment]
Done. Please review.

**Comment by brianhyder** [comment]
@JahlomP @blakecallens What is the status on this?  I'd like to merge it into 0.4.2 early this week (Monday or Tuesday).  Thanks.

**Comment by JahlomP** [comment]
Working on it now

**Comment by JahlomP** [comment]
Fixed #842

**Comment by JahlomP** [comment]
@blakecallens review.

**Comment by blakecallens** [comment]
@JahlomP before I can fully review, I'll need a couple things:

1. This pull request needs to be based off of a fork from our repository. That way I can clone your branch that has the changes and test locally. You'll probably have to reopen a new pull request, but it'll make it much easier for me to review.
2. Can you update this pull request to go with the 0.5.0 branch? We're ready to tag 0.4.2 and I think this is a big enough change that it should go in the next major release.

**Comment by JahlomP** [comment]
OK. I agree.
